catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.metacard.validation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.util.Describable;
+import ddf.catalog.validation.MetacardValidator;
+import ddf.catalog.validation.ValidationException;
+
+public class MetacardValidityMarkerPlugin implements PreIngestPlugin {
+
+    private List<String> enforcedMetacardValidators;
+
+    private List<MetacardValidator> metacardValidators;
+
+    @Override
+    public CreateRequest process(CreateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+
+        // Initialize empty list of metacards to allow through
+        List<Metacard> returnMetacards = new ArrayList<Metacard>();
+        // Loop all metacards in request
+        for (Metacard metacard : input.getMetacards()) {
+            MetacardImpl newMetacard = new MetacardImpl(metacard);
+            // Default to allowing metacard through
+            Boolean toReturn = true;
+            // Run metacard through each validator
+            for (MetacardValidator metacardValidator : metacardValidators) {
```

Comment:
I guess I was picturing the complexity of having to make sure every validator checks it's applicability efficiently and correctly. It's probably not going to be that bad since they'll likely just check the MetacardType in most cases.

I think I've changed my mind, or at least my reasoning. If the validators are all checking the MetacardType anyways, then I still feel that something that repetitive and fundamental should be part of the interface, or maybe service properties, instead. Otherwise it feels hidden in the implementation to me, and makes validating metacards a very all-or-nothing exercise.